CONTRIBUTING.md

```diff
@@ -0,0 +1,194 @@
+# Developer Guidelines
+
+## Code Style
+
+### Existing non-compliant code
+The code that does not comply with these guidelines will be fixed in separate pull request(s). Any new contribution should follow closely these guidelines but should not change style in the existing GridKit code.
+
+### Error handling
+Return values of member functions should be of type `int` and used for error handling. Functions return 0 if no error is encounter, return positive value for warnings and recoverable error, and negative value for irrecoverable errors.
+
+### Output
+If an output is needed (for example, a warning needs to be displayed), use `std::cout` and not `printf` as shown below. There should be a space before and after each `<<`.
```

Comment:
Would there be cases where `printf`'s pretty print would be preferred over `std::cout`?

Reply:
I don't see immediate cases where `printf` would be preferred. If so, we could consider `std::format` per @alexander-novo suggestion. For now, I would say this is outside the scope for this PR. If `printf` becomes needed, I suggest we create separate issue.

Reply:
I am of preference to scientific notation when looking at floats (and displaying them). However, sticking with `std::cout` only is fine for now. If `std::format` is available that would work as well.